tests/scanners/test.javascript.js

```diff
@@ -227,6 +228,45 @@ describe('JavaScript Scanner', function() {
     assert(ok);
   });
 
+  it('should pass addon metadata to rules', () => {
+    var fakeRules = { metadata_not_passed: () => {} };
+
+    sinon.stub(fakeRules, 'metadata_not_passed', (context) => {
+      return {
+        Identifier: (node) => {
+          var metadata = context.settings.addonMetadata;
+
+          if (!metadata || typeof metadata.guid === 'undefined') {
```

Comment:
Is guid guaranteed - it's required but would this mean we'd not be able to carry on with validation even if we wanted to.

Maybe check metadata is an object?